### ISSUE TITLE
Error installing with MacOS 14.0 (Sonoma)

### ISSUE THREAD

**Issue by mikelkew** [eddieantonio/imgcat#51]
When attempting to run `brew install eddieantonio/eddieantonio/imgcat` on MacOS 14.0 (Sonoma), it fails with the following error:

```
Last 15 lines from /Users/mikelkew/Library/Logs/Homebrew/imgcat/01.configure:
__OSX_AVAILABLE(14.0) __IOS_AVAILABLE(17.0) __WATCHOS_AVAILABLE(10.0)
                      ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/term.h:759:17: error: expected parameter declarator
__OSX_AVAILABLE(14.0) __IOS_AVAILABLE(17.0) __WATCHOS_AVAILABLE(10.0)
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/term.h:759:17: error: expected ')'
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/term.h:759:16: note: to match this '('
__OSX_AVAILABLE(14.0) __IOS_AVAILABLE(17.0) __WATCHOS_AVAILABLE(10.0)
               ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/term.h:759:23: error: expected function body after function declarator
__OSX_AVAILABLE(14.0) __IOS_AVAILABLE(17.0) __WATCHOS_AVAILABLE(10.0)
                      ^
6 errors generated.
no
./configure: error: failed to find required header: term.h
```

Have you come across this before, or happen to have any ideas for how to resolve it?

**Comment by eddieantonio** [comment]
I have not tried it on MacOS 14.0 yet. Missing `term.h` seems to be a recurring issue with this repo and the way it specifies dependencies. Quite honestly, I don't know where `term.h` comes from 😅 - it probably has coincidentally been installed on all my machines. I think it's probably part of `ncurses`?

Could you try the following?

    brew install ncurses

**Comment by RafaelChefe** [comment]
Hi 👋 

Tried installing ncurses, still doesn't work... 😞

**Comment by mikelkew** [comment]
Thanks @eddieantonio, unfortunately that's one of the things I tried running from the get-go. No luck 😞

**Comment by eddieantonio** [comment]
I just upgraded to macOS 14.0 and I cannot reproduce this on my machine...  I'll try a few things out. I really want to know where that `term.h` header comes from on macOS!

**Comment by mikelkew** [comment]
Thanks @eddieantonio.. Did you happen to also update the Command Line Tools as well? From the error message I was seeing, it looks like that's where it was trying to get it from:
```
/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/term.h
```

**Comment by eddieantonio** [comment]
Good point. Mine was getting the header from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include

Trying now...

**Comment by eddieantonio** [comment]
Update: `#include <term.h>` _requires_ `#include <unistd.h>`... for some reason... `term.h` is included in the MacOSX14 command line tools SDK. It just requires some weird `#define` statements, and apparently `unistd.h` provides them. 🤷🏽🤷🏽

**Comment by eddieantonio** [comment]
It's because `term.h` relies on macros in `Availability.h`, which provides the macros `__OSX_AVAILABLE`, `__IOS_AVAILABLE`, `__WATCHOS_AVAILABLE`, among others. I suspect that this is an Apple bug, but it was not discovered, because what kind of lunatic would make a C file that has the line `#include <term.h>` and nothing else? (\*points at self\* this guy).

I will push out a workaround shortly.

**Comment by eddieantonio** [comment]
Okay, you should be able to `brew update` && `brew install eddieantonio/eddieantonio/imgcat` and it should work now!

**Comment by mikelkew** [comment]
Thanks @eddieantonio, very much appreciated! 🙏🏼
